store/tikv/prewrite.go

```diff
@@ -140,6 +151,23 @@ func (action actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoff
 					c.run(c, nil)
 				}
 			}
+
+			if c.isOnePC() {
+				if prewriteResp.OnePcCommitTs == 0 {
+					logutil.Logger(bo.ctx).Warn("1pc failed and fallbacks to normal commit procedure",
+						zap.Uint64("satrTS", c.startTS))
+					tikvOnePCTxnCounterFallback.Inc()
+					c.setOnePC(false)
+				} else {
+					c.mu.Lock()
+					if c.onePCCommitTS != 0 {
+						logutil.Logger(bo.ctx).Fatal("one pc happened multiple times",
+							zap.Uint64("startTS", c.startTS))
+					}
+					c.onePCCommitTS = prewriteResp.OnePcCommitTs
+					c.mu.Unlock()
+				}
```

Comment:
And maybe return after this line? If 1pc succeeds, we needn't check for async commit, I think.